### ISSUE TITLE
Use XDG Icon Themes To Facilitate Platform-Native Theming

### ISSUE THREAD

**Issue by elsiehupp** [nextcloud/desktop#3188]
## How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

## Expected behaviour
Nextcloud’s client applications should use platform-native UI icons where possible.

## Actual behaviour
Nextcloud’s client applications use custom UI icons that look equally out-of-place everywhere.

## Client configuration
Client version: 3.x (applies to a wide range of versions, though)
Operating system: All
OS language: All

## Commentary

> I originally posted this as a comment on https://github.com/nextcloud/desktop/issues/2915, but it’s separate enough that it makes sense to repost it as its own Issue, as well.

I’ve been meaning to weigh in on the issues of UI icons (not so much the application icon). If Nextcloud Desktop were to use the [Freedesktop.org Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html) (which Qt natively supports) and the [Freedesktop.org Standard Icon Names](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html), it would facilitate platform-specific UX theming and help the application feel more native.

Linux aside, Microsoft has both [Fluent System Icons](https://github.com/microsoft/fluentui-system-icons) and [Segoe Icons](https://docs.microsoft.com/en-us/windows/uwp/design/style/segoe-ui-symbol-font), while Apple has [SF Symbols](https://developer.apple.com/sf-symbols/). Microsoft’s Fluent Icons are under the MIT License, so if I understand correctly they could be redistributed as part of the Nextcloud client code, but Segoe and SF Symbols would be linked through the operating systems’ APIs due to licensing. In both cases, a translation layer exposing the native icon themes as XDG icon themes would make sense to have as a separately distributable library for the benefit of other application developers, though to be honest that undertaking seems more like something Qt itself should pursue.

On Linux, the main XDG icon themes to target would probably be [Adwaita](https://gitlab.gnome.org/GNOME/adwaita-icon-theme) and [Breeze](https://github.com/KDE/breeze-icons). I don’t know to what extent Adwaita and Breeze use compatible icon names beyond the base specification, though. One nice thing with the XDG icon specification as implemented in both GTK and Qt is that it can specify fallbacks, such that if an icon doesn’t exist in the primary theme a different icon can be used. In this way, Nextcloud Desktop could, for example, include variations of Adwaita and/or Breeze within the distributable so that themable UI icons would have reasonable defaults.

Probably the safest way to start with theming would be to remap the existing icon names onto the Freedesktop.org Icon Naming Specification and port the existing icon theme to the Freedektop.org Icon Theme Specification. Once this is done, Nextcloud Desktop could start adding whitelisted platform-specific icon themes that can be thoroughly tested. Ultimately allowing arbitrary icon themes seems like it should be a configurable option for the Linux version, but it should be disabled by default.

If any major Linux distributions that don’t use Adwaita or Breeze by default want Nextcloud to natively support their icon theme, it would be relatively easy for the distribution’s maintainers to make sure that their theme includes all of the necessary icons and then add the theme to Nextcloud’s whitelist.

### Port the existing theme to Qt’s XDG icon-theme interface
- [ ] Translate existing icon names to XDG-corresponding values (using the icon lists from Adwaita and Breeze when the XDG standard lacks specificity)
- [ ] Create an XDG-compliant `.theme` for the existing icon theme
- [ ] Change icon imports in the code to use Qt’s XDG icon-theme interface

### Add Native Themes
- [ ] Test Adwaita and Breeze
- [ ] Create an interface for Fluent or Segoe
- [ ] Create an interface for SF Symbols

**Comment by elsiehupp** [comment]
Look, a progress indicator!

<img width="665" alt="github issue progress indicator" src="https://user-images.githubusercontent.com/9206310/115757952-9c5bf600-a36c-11eb-9738-ada638cfbbec.png">

**Comment by elsiehupp** [comment]
I should also mention that using XDG icon themes would facilitate further OEM interface theming, should that be desired.

**Comment by Rello** [comment]
Hello,

due to the recent and ongoing Qt efforts, I am closing this one...
